lib/ansible/modules/cloud/amazon/cloudformation.py

```diff
@@ -351,6 +368,24 @@ def update_stack(module, stack_params, cfn):
     return result
 
 
+def update_termination_protection(module, stack_params, cfn):
+    '''updates termination protection of a stack'''
+    stack = get_stack_facts(cfn, stack_params['StackName'])
+    if stack:
+        if stack['EnableTerminationProtection'] is not module.params['termination_protection']:
```

Comment:
I think, unless you have a good reason, that you should use stack_params here and four lines down instead of module.params. Honestly I wish we didn't have to push module around.

Reply:
Passing in stack_name and the desired state of 'termination protection' instead of looking into module.params inside the function.
Also, since I am using `exit_json` and `fail_json`, I need `module`.

Reply:
Yeah, I know you need module. That was just a general musing.